scripts/one-off/records-migration/main.ts

```diff
@@ -0,0 +1,171 @@
+import Knex from 'knex';
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+const SOURCE_DB_SCHEMA = 'nango';
+const SOURCE_DB_TABLE = '_nango_sync_data_records';
+const sourceKnex = Knex({
+    client: 'pg',
+    connection: {
+        host: '*****',
+        port: 5432,
+        database: '*****',
+        user: '*****',
+        password: '*****',
+        ssl: true,
+        statement_timeout: 60000
+    },
+    pool: {
+        min: 3,
+        max: 6
+    }
+});
+
+const TARGET_DB_SCHEMA = 'nango_records';
+const TARGET_DB_TABLE = 'records';
+const targetKnex = Knex({
+    client: 'pg',
+    // TODO:
+    connection: {
+        host: 'localhost',
+        port: 5432,
+        database: 'nango',
+        user: 'nango',
+        password: 'nango',
+        ssl: false,
+        statement_timeout: 60000
+    },
+    pool: {
+        min: 3,
+        max: 6
+    }
+});
+
+interface Checkpoint {
+    lastCreatedAt: string | null;
+    lastId: string | null;
+}
+
+interface TargetRecord {
+    id: string;
+    external_id: string;
+    json: object;
+    data_hash: string;
+    connection_id: number;
+    model: string;
+    sync_id: string;
+    sync_job_id: number;
+    created_at_raw: string;
+    created_at: string;
+    updated_at: string;
+    deleted_at: string;
+}
+
+const dirname = path.dirname(fileURLToPath(import.meta.url));
+const checkpointFile = path.join(dirname, 'migration_checkpoint.json');
+
+async function migrateData() {
+    console.log('Starting data migration...');
+    const batchSize = 1000;
+    let checkpoint = await getCheckpoint();
+
+    let more = true;
+    while (more) {
+        const startRead = Date.now();
+        const records = (await sourceKnex
+            .select(
+                sourceKnex.raw(`
+                    id,
+                    external_id,
+                    json,
+                    data_hash,
+                    nango_connection_id as connection_id,
+                    model,
+                    sync_id,
+                    sync_job_id,
+                    to_json(created_at) as created_at,
+                    to_json(updated_at) as updated_at,
+                    to_json(external_deleted_at) as deleted_at,
+                    created_at as created_at_raw
+                `)
+            )
+            .from<TargetRecord>(`${SOURCE_DB_SCHEMA}.${SOURCE_DB_TABLE}`)
+            .where((builder) => {
+                if (checkpoint.lastCreatedAt && checkpoint.lastId) {
+                    builder.where(sourceKnex.raw(`(created_at, id) > (?, ?)`, [checkpoint.lastCreatedAt, checkpoint.lastId]));
+                }
+            })
+            .orderBy([
+                { column: 'created_at_raw', order: 'asc' },
+                { column: 'id', order: 'asc' }
+            ])
+            .limit(batchSize)) as unknown as TargetRecord[];
```

Comment:
no need to change for a script but to avoid this you need to set `select<TargetRecord[]>`